sql/core/src/main/scala/org/apache/spark/sql/execution/command/ddl.scala

```diff
@@ -485,6 +485,7 @@ case class AlterTableAddPartitionCommand(
       catalog.createPartitions(table.identifier, batch, ignoreIfExists = ifNotExists)
     }
 
+    sparkSession.catalog.refreshTable(table.identifier.quotedString)
```

Comment:
just curious, does it matter whether we refresh cache before or after the stats are updated?

Reply:
In this particular case, it doesn't matter because table size is re-calculated by getting file statuses directly from the filesystem. So, the cached data is not used in updating table stats.

I think we should review other places.

Just in case, I will update the test and check that table size is updated after adding of the partition.

Reply:
> Just in case, I will update the test and check that table size is updated after adding of the partition.

Let me add such checking later independently from this PR. I think I have found one more issue relating to looking for `HiveTableRelation` in the cache of Cache Manager.

Reply:
+1 for @MaxGekk 's decision.

Reply:
Yup, SGTM :)

Reply:
Here is the bug fix: https://github.com/apache/spark/pull/31112 . Updating of table stats triggers the bug.

Reply:
> Let me add such checking later independently from this PR.

I added the check for partition adding as well. Please, review this PR: https://github.com/apache/spark/pull/31131